pkg/controller/controller.go

```diff
@@ -501,6 +502,11 @@ func (c *Controller) runAction(ctx context.Context, t *tomb.Tomb, as *crv1alpha1
 				msg = fmt.Sprintf("Failed to init phase params: %#v:", as.Status.Actions[aIDX].Phases[i])
 			}
 
+			if err != nil {
+				data, _ := json.Marshal(err)
+				err = errkit.New(string(data))
+			}
```

Comment:
It doesn't feel to me as a proper solution here. `err` already contains (should contain the fields). By doing this, we create an json-like string, which is not easily analyzable. See below my proposal.

Reply:
instead of calling `err.Error()` we can do this:

1. Declare interface:
```
type errorWithDetails interface {
	error
	Details() errkit.ErrorDetails
}
```

2. Assert the error:
```
var ewd errorWithDetails
if errors.As(err, &ewd) {
	details := ewd.Details()
	// Use details
} else {
	// Not an error with details
}
```

In details you will have `map[string]any` that you will be able to marshal to json and append to the error message you get using .Error() method.